offset.go

```diff
@@ -0,0 +1,414 @@
+		flags:     offsetFlags(),
+		parseArgs: offsetParseArgs,
+		run:       offsetRun,
+	}
+}
+
+func offsetFlags() *flag.FlagSet {
+	offset := flag.NewFlagSet("offset", flag.ExitOnError)
+	offset.StringVar(&config.offset.args.brokers, "brokers", "", "Comma separated list of brokers. Port defaults to 9092 when omitted.")
+	offset.StringVar(&config.offset.args.group, "group", "", "The name of the consumer group.")
+	offset.StringVar(&config.offset.args.topic, "topic", "", "The full or partial name of topic(s)")
+	offset.IntVar(&config.offset.args.partition, "partition", -1, "The identifier of the partition")
+	offset.StringVar(&config.offset.args.set, "setConsumerOffsets", "", "Set offsets for the consumer groups to \"oldest\", \"newest\", or a specific numerical value. "+
+		"For more accurate modification also specify topic and/or partition")
+	offset.BoolVar(&config.offset.args.verbose, "verbose", false, "More verbose logging to stderr.")
+	offset.StringVar(&config.offset.args.version, "version", "", "Kafka protocol version")
+
+	offset.Usage = func() {
+		fmt.Fprintln(os.Stderr, "Usage of offset:")
+		offset.PrintDefaults()
+		fmt.Fprintln(os.Stderr, `
+The values for -brokers can also be set via the environment variable KT_BROKERS respectively.
+The values supplied on the command line win over environment variable values.
+
+Offsets are listed for partitions/consumer groups. This can be filtered by the flags [brokers], [topic], and [partition].
+
+
+This tool also offers the option to update these by adding the [setConsumerOffsets] flag.
+
+The following syntax is supported for setConsumerOffsets:
+
+  (oldest|newest)?(\d+)?
+
+ - "oldest" and "newest" refer to the oldest and newest offsets known for a
+   given partition.
+
+ - Given only a numeric value, it is interpreted as an absolute offset value.
+`)
+		os.Exit(2)
+	}
+
+	return offset
+}
+
+func offsetParseArgs() {
+	envBrokers := os.Getenv("KT_BROKERS")
+	if config.offset.args.brokers == "" {
+		if envBrokers != "" {
+			config.offset.args.brokers = envBrokers
+		} else {
+			config.offset.args.brokers = "localhost:9092"
+		}
+	}
+	config.offset.brokers = strings.Split(config.offset.args.brokers, ",")
+	for i, b := range config.offset.brokers {
+		if !strings.Contains(b, ":") {
+			config.offset.brokers[i] = b + ":9092"
+		}
+	}
+
+	re, err := regexp.Compile(config.offset.args.topic)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Invalid regex for filter. err=%s\n", err)
+		os.Exit(2)
+	}
+	config.offset.topic = re
+
+	config.offset.partition = int32(config.offset.args.partition)
+
+	config.offset.group = config.offset.args.group
+
+	if config.offset.args.set == "" {
+		config.offset.set = -1
+	} else if config.offset.args.set == "newest" {
+		config.offset.setNewest = true
+	} else if config.offset.args.set == "oldest" {
+		config.offset.setOldest = true
+	} else if newOffset, err := strconv.Atoi(config.offset.args.set); err == nil {
+		config.offset.set = int64(newOffset)
```

Comment:
I looked at this, but the sarama constants have int values of -1 and -2.

If you create a new consumer group its default offset for a topic is -1, so someone might try to set that value in order to reset a group offset. 
This would conflict with the constant, because -1 equals the constant sarama.OffsetNewest (the opposite of what someone might be trying to achieve)